src/fromager/constraints.py

```diff
@@ -12,8 +12,36 @@
 
 
 class Constraints:
-    def __init__(self, data: dict[str, Requirement]):
-        self._data = {canonicalize_name(n): v for n, v in data.items()}
+    def __init__(self) -> None:
+        # mapping of canonical names to requirements
+        # NOTE: Requirement.name is not normalized
+        self._data: dict[NormalizedName, Requirement] = {}
+
+    def __iter__(self) -> typing.Iterable[NormalizedName]:
+        yield from self._data
+
+    def add_constraints(self, *reqs: str) -> None:
```

Comment:
What's the limit on the number of arguments that can be passed to a function? I wouldn't expect constraints files to be very long, but does this constrain them in a way that using an iterable as argument would not?

Reply:
Python does not have an argument limit. Arguments are only limited by amount of memory.

The method doesn't have to accept an arbitrary amount of arguments. I have changed it to accept a single requirement string. Caller can call the method in a loop.